### PR TITLE
Feature sb 100 products requests

### PR DESCRIPTION
**Pull request by likeri29**
1. I updated the product service and added sorting options there (by date).
2. I added a new admin controller for handling products on the admin page (documentation regarding this controller is on the second screen)
3. I wrote tests that cover the product service

[JIRA task](https://team-jfqe9wlknaql.atlassian.net/jira/software/c/projects/SB/boards/8?assignee=712020%3Abc32ac79-d2de-4a3e-accf-51396ebf73bb&selectedIssue=SB-100)

Screenshot 2024-07-08 at 16 11 15

Screenshot 2024-07-09 at 11 53 20

### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet
### Frontend
- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [x] Unit tests written
- [x] use ConfigService instead of process.env
- [x] use transactions if there is a call chain that mutates data in different tables
- [x] use @index decorator for frequently requested data

### CODE REVIEW

File: src/modules/products/products.service.ts [hunk 1]
```diff
@@ -12,8 +13,16 @@ export class ProductsService {
...
+  async findAll(query: GetProductsQueryDto): Promise<{
+    products: ProductEntity[];
+    totalCount: number;
+    totalPages: number;
+  }> {
+    const { products, totalCount } =
+      await this.productsRepository.findAll(query);
+    const limit = query.limit || PRODUCTS_LIMIT;
+    const totalPages = Math.ceil(totalCount / limit);
+    return { products, totalCount, totalPages };
```
Comment (VladManukian) [inline src/modules/products/products.service.ts]:
```suggestion
    const totalPages = Math.ceil(totalCount / limit);
    
    return { products, totalCount, totalPages };
```

File: src/modules/admin/admin.controller.ts [hunk 2]
```diff
@@ -63,4 +69,13 @@ export class AdminController {
       limit,
     );
   }
+
+  @Get('products')
```
Comment (lyudad) [inline src/modules/admin/admin.controller.ts:75]: @ApiOperation

File: src/modules/admin/admin.controller.ts [hunk 3]
```diff
@@ -13,17 +13,22 @@ import {
...
 @Controller('admin')
 @UseGuards(RolesGuard)
 export class AdminController {
-  constructor(private readonly usersService: UsersService) {}
+  constructor(
+    private readonly usersService: UsersService,
+    private readonly productsService: ProductsService,
+  ) {}
 
   @Get('user/:id')
```
Comment (lyudad) [inline src/modules/admin/admin.controller.ts:35]: @ApiOperation({
    summary: '...',
    description: '...',

Comment (likeri29) [reply src/modules/admin/admin.controller.ts:35]: Added api tag for the controller

File: src/modules/admin/admin.controller.ts (hunk 2 above)
Comment (likeri29) [reply src/modules/admin/admin.controller.ts:75]: Added description and summary for product route. How it looks on swagger you can see on the screenshot above

**Review by lyudad — APPROVED** [review]

**Pull request merged**